onnxruntime/core/providers/webgpu/webgpu_execution_provider.cc

```diff
@@ -435,7 +435,7 @@ std::unique_ptr<KernelRegistry> RegisterKernels() {
       KERNEL_CREATE_INFO(9, Cosh),
       KERNEL_CREATE_INFO(9, Asinh),
       KERNEL_CREATE_INFO(9, Acosh),
-      KERNEL_CREATE_INFO(9, Atanh),
+      // KERNEL_CREATE_INFO(9, Atanh),  TEMPORARY - Doesn't handle 1.0f -> inf with Metal
```

Comment:
how do we want to deal with Atanh on macOS/iOS? add a `#ifdef` for apple?

Reply:
If there any way to do it gracefully in the shader?

Reply:
what I can think is using `#ifdef APPLE` together with shader string generation.

for non-apple: just use builtin atanh
for apple: use custom_atanh:

fn custom_atanh(a) {
  if (a == 1.0f) {
       ...
  }
  return atanh(a);
}

Reply:
Not sure it's possible.

https://www.w3.org/TR/WGSL/#differences-from-ieee754
> A [shader-creation error](https://www.w3.org/TR/WGSL/#shader-creation-error) results if any [const-expression](https://www.w3.org/TR/WGSL/#const-expressions) of floating-point type overflows or evaluates to NaN or infinity.

We'd need to return +infinity, but it sounds like that's not possible.